test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1724,6 +1724,79 @@ UCS_TEST_P(test_ucp_sockaddr_protocols, am_zcopy_64k,
 UCP_INSTANTIATE_CM_TEST_CASE(test_ucp_sockaddr_protocols)
 
 
+class test_ucp_sockaddr_protocols_diff_config : public test_ucp_sockaddr_protocols
+{
+public:
+    void init() {
+        if (is_self()) {
+            UCS_TEST_SKIP_R("self - same config");
+        }
+
+        m_err_count = 0;
+        get_sockaddr();
+        test_base::init();
+    }
+
+    void init_entity(const char *num_paths) {
+        /* coverity[tainted_string_argument] */
+        ucs::scoped_setenv num_paths_env("UCX_IB_NUM_PATHS", num_paths);
+        create_entity();
+    }
+
+    void create_entities_and_connect(bool server_less_num_paths) {
+        /* coverity[tainted_string_argument] */
+        ucs::scoped_setenv max_eager_lanes_env("UCX_MAX_EAGER_LANES", "2");
+
+        if (server_less_num_paths) {
+            // create the client
+            init_entity("2");
+            // create the server
+            init_entity("1");
+        } else {
+            // create the client
+            init_entity("1");
+            // create the server
+            init_entity("2");
+        }
+
+        skip_loopback();
```

Comment:
duplication of 1731-1733?

Reply:
you mean if skip_loopback is the same as is_self?
&sender() == &receiver() can be true for any transport, not only for self.

Reply:
i see
but in this test, seems it's not possible sender==receiver since you create 2 entities?